test/e2e/nodelatencymonitor_test.go

```diff
@@ -0,0 +1,210 @@
+
+	t.Log("Latency probes are confirmed to be enabled.")
+}
+
+func testDisableLatencyProbes(t *testing.T, data *TestData) {
+
+	_, err := data.crdClient.CrdV1alpha1().NodeLatencyMonitors().Get(context.TODO(), nodeLatencyMonitorName, v1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			t.Logf("NodeLatencyMonitor CR does not exist, no need to delete.")
+			return
+		}
+		t.Fatalf("Error when checking for NodeLatencyMonitor CR: %v", err)
+	}
+
+	err = data.crdClient.CrdV1alpha1().NodeLatencyMonitors().Delete(context.TODO(), nodeLatencyMonitorName, v1.DeleteOptions{})
+	if err != nil {
+		t.Fatalf("Failed to delete NodeLatencyMonitor CR: %v", err)
+	}
+	t.Logf("Latency probes are confirmed to be disabled..")
+}
+
+func testUpdatePingInterval(t *testing.T, data *TestData) {
+	nlm, err := data.crdClient.CrdV1alpha1().NodeLatencyMonitors().Get(context.TODO(), nodeLatencyMonitorName, v1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			t.Logf("NodeLatencyMonitor CR not found, creating a new one.")
+			pingInterval := int32(60)
+			createOrUpdateNodeLatencyMonitorCR(t, data, pingInterval)
+			nlm, err = data.crdClient.CrdV1alpha1().NodeLatencyMonitors().Get(context.TODO(), nodeLatencyMonitorName, v1.GetOptions{})
+			if err != nil {
+				t.Fatalf("Failed to retrieve NodeLatencyMonitor CR after creation: %v", err)
+			}
+		} else {
+			t.Fatalf("Failed to retrieve NodeLatencyMonitor CR: %v", err)
+		}
+	}
+
+	// Update the ping interval
+	newInterval := generatePingInterval()
+	nlm.Spec.PingIntervalSeconds = newInterval
+	_, err = data.crdClient.CrdV1alpha1().NodeLatencyMonitors().Update(context.TODO(), nlm, v1.UpdateOptions{})
+	if err != nil {
+		t.Fatalf("Failed to update NodeLatencyMonitor CR: %v", err)
+	}
+
+	t.Logf("Ping interval updated successfully to %d seconds.", newInterval)
+}
+
+func testRetrieveLatencyStats(t *testing.T, data *TestData) {
+
+	summary := &statsv1alpha1.NodeLatencyStats{
+		ObjectMeta: v1.ObjectMeta{Name: "kind-worker"},
+		PeerNodeLatencyStats: []statsv1alpha1.PeerNodeLatencyStats{
+			{
+				NodeName: "kind-control-plane",
+				TargetIPLatencyStats: []statsv1alpha1.TargetIPLatencyStats{
+					{
+						LastMeasuredRTTNanoseconds: 5837000,
+						LastRecvTime:               v1.Time{Time: time.Now().Add(-time.Minute)},
+						LastSendTime:               v1.Time{Time: time.Now().Add(-time.Minute)},
+						TargetIP:                   "10.10.0.1",
+					},
+				},
+			},
+			{
+				NodeName: "kind-worker2",
+				TargetIPLatencyStats: []statsv1alpha1.TargetIPLatencyStats{
+					{
+						LastMeasuredRTTNanoseconds: 4704000,
+						LastRecvTime:               v1.Time{Time: time.Now().Add(-time.Minute)},
+						LastSendTime:               v1.Time{Time: time.Now().Add(-time.Minute)},
+						TargetIP:                   "10.10.2.1",
+					},
+				},
+			},
+		},
+	}
+
+	ctx := context.Background()
+
+	_, err := data.crdClient.StatsV1alpha1().NodeLatencyStats().Create(ctx, summary, v1.CreateOptions{})
```

Comment:
there is a misunderstanding here. It is Antrea that should create the `NodeLatencyStats` object as a result of enabling the NodeLatencyMonitor functionality. The test case should not be creating it.

Reply:
yes, i have some confusion here. 
can you provide me documentation of it

Reply:
Documentation for the feature is available here: https://github.com/antrea-io/antrea/blob/main/docs/feature-gates.md#nodelatencymonitor

The NodeLatencyMonitor CRD is to enable ICMP probes, which will measure latency between pairs of Nodes. The latency measurements are then reported by Antrea through the NodeLatencyStats API. The e2e test should validate that when enabling NodeLatencyMonitor, invoking the NodeLatencyStats API (GET) returns appropriate measurements.

Reply:
One thing to clarify, 
My implementation is almost right other than the creation part?

Or the whole function is wrong ?